server/components/fill_teams_test.js

```diff
@@ -0,0 +1,195 @@
+  //   }).then(function() {
+  //     return setUpDatabase();
+  //   }).then(function() {
+  //     return findTeams();
+  //   }).then(function() {
+  //     return assignPlayers(selectedTeams, selectedPlayers)
+  //   });
+  // }
+
+  var _fillTeams = function(user) {
+    if (user.role !== 'admin') {
+      console.log("Error, user is not an admin, user role is: " + user.role);
+      return Promise.reject("User is not an admin");
+    }
+    db.Sport.bulkCreate(stockSport).then(function() {
+      return db.Team.bulkCreate(stockTeams);
+    }).then(function() {
+      return db.Player.bulkCreate(stockPlayers);
+    }).then(function() {
+      return db.Sport.find(1)
+    }).then(function(sport) {
+      return setCurrentSport(sport);
+    }).then(function() {
+      return db.Team.findAll({
+        where: [{special: null, SportId: currentSport.id}]
+      });
+    }).then(function(teams) {
+      return setSelectedTeams(teams);
+    }).then(function() {
+      return db.Player.findAll({
+        where: {SportId: currentSport.id}
+      });
+    }).then(function(players) {
+      return setSelectedPlayers(players);
+    }).then(function() {
+      return assignPlayers(selectedTeams, selectedPlayers);
+    });
+  }
+
+  // var userRoleCheck = function(user) {
+  //   if (user.role !== 'admin') {
+  //     console.log("Error, user is not an admin, user role is: " + user.role);
+  //     return Promise.reject("User is not an admin");
+  //   }
+  // }
+
+  // var setUpDatabase = function() {
+  //   db.Sport.bulkCreate(stockSport).then(function() {
+  //     return db.Team.bulkCreate(stockTeams);
+  //   }).then(function() {
+  //     return db.Player.bulkCreate(stockPlayers);
+  //   });
+  // }
+
+  // var findTeams = function() {
+  //   db.Sport.find(1).then(function(sport) {
+  //     return setCurrentSport(sport);
+  //   }).then(function() {
+  //     return db.Team.findAll({
+  //       where: [{special: null, SportId: currentSport.id}]
+  //     });
+  //   }).then(function(teams) {
+  //     return setSelectedTeams(teams);
+  //   }).then(function() {
+  //     return db.Player.findAll({
+  //       where: {SportId: currentSport.id}
+  //     });
+  //   }).then(function(players) {
+  //     return setSelectedPlayers(players);
+  //   });
+  // }
+
+  var setCurrentSport = function(sport) {
+    currentSport = sport;
+  }
+
+  var setSelectedTeams = function(teams) {
+    selectedTeams = teams;
+  }
+
+  var setSelectedPlayers = function(players) {
+    selectedPlayers = players;
+  }
+
+
+  var assignPlayers = function(teams, players) {
+    getRealWorldTeams(players);
+    _(realWorldTeams).map(function(realTeam) {
+      var realTeamIndex = realWorldTeams.indexOf(realTeam);
+      db.Player.findAll({
+       where: { realWorldTeam: realTeam }
+      }).then(function(realTeamPlayers) {
+        _(realTeamPlayers).map(function(realTeamPlayer) {
+          if (realTeamIndex >= teams.length) {
+            db.Team.find({
```

Comment:
Do note that in starting a new promise chain here, any errors that happen in the internal promise chain won't be caught in the higher promise chain. You may consider using Promise.map instead of "_.map" in this loop.